cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/bulkwriter/StreamSession.java

```diff
@@ -169,7 +169,9 @@ List<RingInstance> getReplicas()
         List<RingInstance> replicasForTokenRange = overlappingRanges.values().stream()
                                                                     .flatMap(Collection::stream)
                                                                     .distinct()
-                                                                    .filter(instance -> !exclusions.contains(instance))
+                                                                    .filter(instance -> !isExclusion(instance,
+                                                                                                     failedInstances,
+                                                                                                     tokenRangeMapping.getBlockedInstances()))
```

Comment:
Filter will run the lambda every time for every instance, so I would suggest computing it once before running the stream
```suggestion
        Set<String> blockedInstances = tokenRangeMapping.getBlockedInstances();
        ....
                                                                    .filter(instance -> !isExclusion(instance,
                                                                                                     failedInstances,
                                                                                                     blockedInstances))
```

Reply:
Makes sense. I assumed `getBlockedInstances` was a lookup on a precomputed value. Will update